### PR TITLE
Add new configuration RPC endpoints

### PR DESCRIPTION
**Pull request by MexicanAce**
# What :computer: 
* Add new configuration RPC endpoints (`config_getShowCalls`, `config_setShowCalls`, and `config_setResolveHashes`)
* Add example http REST file for quickly testing the node locally
* Add recommended plugins for VSCode

# Why :hand:
* These endpoints help update the configuration of stream output from the in-memory node without requiring a restart of the node (potentially losing information)
* Having the `.http` file has made an easy way for developers to make requests to the available RPC endpoints without memorizing or copy/pasting long CURL commands
* Recommended plugins will help developers use `.http` files and setup their local environment faster

# Evidence :camera:
In this video, I start a node with just `era_test_node run` and update the `--show-calls` and `--resolve-hashes` parameters via RPC calls while the node is still running:

https://github.com/matter-labs/era-test-node/assets/1890113/89d3f378-b559-4257-b728-7b025e9341ee

# Notes :memo:
* I'm open to changing the name of these endpoints as `config_` was just the start.
* I really REALLY think this will be super useful when we get to the point of allowing end users to dynamically configure things like the delta of `current_timestamp` or the gas price (currently hard-coded to 0.25 gwei)

### CODE REVIEW

**Review by uF4No — APPROVED** [review]
LGTM.
Awesome feature to spin up the node and only enable traces when needed without losing the state 🤘

File: src/configuration_api.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#[rpc]
+pub trait ConfigurationApiNamespaceT {
+    #[rpc(name = "config_getShowCalls", returns = "String")]
+    fn config_get_show_calls(&self) -> Result<String>;
+    
+    #[rpc(name = "config_setShowCalls", returns = "String")]
+    fn config_set_show_calls(&self, value: String) -> Result<String>;
+
+    #[rpc(name = "config_setResolveHashes", returns = "bool")]
+    fn config_get_resolve_hashes(&self, value: bool) -> Result<bool>;
```
Comment (mm-zk) [inline src/configuration_api.rs:32]: nit: 
name is called 'get' and the RPC name is 'set'

File: src/configuration_api.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+impl ConfigurationApiNamespace {
+    pub fn new(node: Arc<RwLock<InMemoryNodeInner>>) -> Self {
+        Self { node }
+    }
+}
+
+#[rpc]
+pub trait ConfigurationApiNamespaceT {
+    #[rpc(name = "config_getShowCalls", returns = "String")]
```
Comment (mm-zk) [inline src/configuration_api.rs:25]: please add short comments about all these 3 methods

File: src/main.rs [hunk 3]
```diff
@@ -119,6 +132,18 @@ pub enum ShowCalls {
     All,
 }
 
+impl ShowCalls {
+    pub fn from_str(s: &str) -> Option<Self> {
+        match s {
```
Comment (mm-zk) [inline src/main.rs:137]: I'd do a to_lowercase() -- so that we also accept when people send 'none' or 'all'

File: src/configuration_api.rs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+impl ConfigurationApiNamespaceT for ConfigurationApiNamespace {
+    fn config_get_show_calls(&self) -> Result<String> {
+        let reader = self.node.read().unwrap();
+        Ok(reader.show_calls.to_string())
+    }
+
+    fn config_set_show_calls(&self, value: String) -> Result<String> {
+        let show_calls = ShowCalls::from_str(&value);
+        
+        match show_calls {
```
Comment (mm-zk) [inline src/configuration_api.rs:44]: you can refactor this a little:
```
let mut inner = self.nolde.write().unwrap();
if let Some(show_calls) = show_calls {
  inner.show_calls = show_calls;
} 
Ok(reader.show_calls.to_string());
```

File: src/configuration_api.rs [hunk 5]
```diff
@@ -0,0 +1,62 @@
+// Built-in uses
+use std::sync::{Arc, RwLock};
```
Comment (mm-zk) [inline src/configuration_api.rs:2]: let's also update the README file, and add some 'curl' / 'http' commands that people can copy & run.

**Review by mm-zk — APPROVED** [review]

File: src/main.rs (hunk 3 above)
Comment (MexicanAce) [reply src/main.rs:137]: Ah good call

File: src/configuration_api.rs (hunk 4 above)
Comment (MexicanAce) [reply src/configuration_api.rs:44]: Oh nice!! I tried to do something similar but having the reader in the same path as the write caused errors with Rust.

File: src/configuration_api.rs (hunk 5 above)
Comment (MexicanAce) [reply src/configuration_api.rs:2]: Sounds good. I'll add a table listing these JSON RPC commands that are added and examples on how to execute them.

**Comment by dutterbutter** [comment]
This is great! Can you please make a PR to reflect these changes to the [docs](https://era.zksync.io/docs/tools/testing/era-test-node.html#sending-network-calls) as well.

File: src/configuration_api.rs (hunk 4 above)
Comment (MexicanAce) [reply src/configuration_api.rs:44]: Yup, this runs into the same deadlock error as I hit before:
`hread 'tokio-runtime-worker' panicked at 'rwlock read lock would result in deadlock'`

File: src/configuration_api.rs (hunk 5 above)
Comment (MexicanAce) [reply src/configuration_api.rs:2]: Added a basic `SUPPORTED_APIS.md` file for now

File: SUPPORTED_APIS.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
+# 🔧 Supported APIs for In-Memory Node 🔧
+
+> ⚠️ **WORK IN PROGRESS**: This list is non-comprehensive and being updated
+
```
Comment (uF4No) [inline SUPPORTED_APIS.md:4]: Just an idea: we could add a summary table on top with just the RPC method, if it's implemented and, maybe in the future, a link to a GH issue so others can collaborate on this.

Comment (MexicanAce) [reply SUPPORTED_APIS.md:4]: Yeah, kinda like a table of contents, I like it 😄 .

I definitely want to link to a GH Issue for each API that's not supported, but I have a follow-up in the coming days to fill out the APIs in this document and I will include that change as part of that PR.

**Review by idea404 — APPROVED** [review]
LGTM. I second @uF4No's idea on using a table on the markdown file to track and quickly navigate endpoints. Let's discuss what the endpoints are that are needed

**Pull request merged**